tests/test_decision_making/test_utility_functions/test_utility_functions.py

```diff
@@ -0,0 +1,158 @@
+    [
+        (ProbabilityOfImprovement, {}),
+        (ThompsonSampling, {"num_features": 100}),
+    ],
+)
+def test_utility_function_no_objective_dataset_raises_error(
+    utility_function_builder: Type[AbstractSinglePointUtilityFunctionBuilder],
+    utility_function_kwargs: dict,
+):
+    key = jr.key(42)
+    forrester = Forrester()
+    dataset = forrester.generate_dataset(num_points=10, key=key)
+    posterior = generate_dummy_conjugate_posterior(dataset)
+    posteriors = {OBJECTIVE: posterior}
+    datasets = {"CONSTRAINT": dataset}
+    with pytest.raises(ValueError):
+        utility_function = utility_function_builder(**utility_function_kwargs)
+        utility_function.build_utility_function(
+            posteriors=posteriors, datasets=datasets, key=key
+        )
+
+
+@pytest.mark.filterwarnings(
+    "ignore::UserWarning"
+)  # Sampling with tfp causes JAX to raise a UserWarning due to some internal logic around jnp.argsort
+@pytest.mark.parametrize(
+    "utility_function_builder, utility_function_kwargs",
+    [
+        (ProbabilityOfImprovement, {}),
+        (ThompsonSampling, {"num_features": 100}),
+    ],
+)
+def test_non_conjugate_posterior_raises_error(
+    utility_function_builder: Type[AbstractSinglePointUtilityFunctionBuilder],
+    utility_function_kwargs: dict,
+):
+    key = jr.key(42)
+    forrester = Forrester()
+    dataset = forrester.generate_dataset(num_points=10, key=key)
+    posterior = generate_dummy_non_conjugate_posterior(dataset)
+    posteriors = {OBJECTIVE: posterior}
+    datasets = {OBJECTIVE: dataset}
+    with pytest.raises(ValueError):
+        utility_function = utility_function_builder(**utility_function_kwargs)
+        utility_function.build_utility_function(
+            posteriors=posteriors, datasets=datasets, key=key
+        )
+
+
+@pytest.mark.parametrize(
+    "utility_function_builder, utility_function_kwargs",
+    [
+        (ProbabilityOfImprovement, {}),
+        (ThompsonSampling, {"num_features": 100}),
+    ],
+)
+@pytest.mark.parametrize(
+    "test_target_function",
+    [(Forrester()), (LogarithmicGoldsteinPrice())],
+)
+@pytest.mark.parametrize("num_test_points", [50, 100])
+@pytest.mark.parametrize("key", [jr.key(42), jr.key(10)])
+@pytest.mark.filterwarnings(
+    "ignore::UserWarning"
+)  # Sampling with tfp causes JAX to raise a UserWarning due to some internal logic around jnp.argsort
+def test_utility_functions_have_correct_shapes(
+    utility_function_builder: Type[AbstractSinglePointUtilityFunctionBuilder],
+    utility_function_kwargs: dict,
+    test_target_function: AbstractContinuousTestFunction,
+    num_test_points: int,
+    key: KeyArray,
+):
+    dataset = test_target_function.generate_dataset(num_points=10, key=key)
+    posterior = generate_dummy_conjugate_posterior(dataset)
+    posteriors = {OBJECTIVE: posterior}
+    datasets = {OBJECTIVE: dataset}
+    ts_utility_builder = utility_function_builder(**utility_function_kwargs)
```

Comment:
nit: Some of the names of variables in this file e.g. `ts_utility_builder` could do with updating now that these tests are more "utility function agnostic".

Reply:
Solved!